ui/browser/views/navbar/location.jsx

```diff
@@ -20,15 +20,52 @@ import { fixURL, getCurrentWebView } from '../../browser-util';
  * to ipcRenderer on mount, and because the key-down event handler is non
  * trivial.
  */
-class Location extends Component {
-  constructor() {
-    super();
-    this.handleKeyDown = this.handleKeyDown.bind(this);
-  }
+const Location = createClass({
+
+  propTypes: {
+    page: PropTypes.object.isRequired,
+    onLocationChange: PropTypes.func.isRequired,
+    onLocationContextMenu: PropTypes.func.isRequired,
+    onLocationReset: PropTypes.func.isRequired,
+    bookmark: PropTypes.func.isRequired,
+    unbookmark: PropTypes.func.isRequired,
+    ipcRenderer: PropTypes.object.isRequired,
+  },
+
+  getInitialState() {
+    return {
+      showURLBar: false,
+    };
+  },
 
   componentDidMount() {
     this.props.ipcRenderer.on('focus-urlbar', () => this.refs.input.select());
-  }
+  },
+
+  componentDidUpdate() {
+    // If we're showing the URL bar, it should be focused. The scenario
+    // where this isn't true is immediately after displaying the URL bar,
+    // so give it focus
+    if (this.state.showURLBar && document.activeElement !== this.refs.input) {
```

Comment:
Yeah, I don't like this either. Or if there's way to determine if an element has focus without the document